controllers/pipeline/pipeline_adapter.go

```diff
@@ -197,7 +199,7 @@ func (a *Adapter) EnsureSnapshotPassedAllTests() (reconciler.OperationResult, er
 		if compositeSnapshot != nil {
 			a.logger.Info("The global component list has changed in the meantime, marking snapshot as Invalid",
 				"snapshot.Name", existingSnapshot.Name)
-			gitops.SetSnapshotIntegrationStatusAsInvalid(existingSnapshot,
+			gitops.SetSnapshotIntegrationStatusAsInvalidandNoRequeue(existingSnapshot,
```

Comment:
I'm thinking that we might want to introduce a new status condition in order to separate invalid Snapshots from Snapshots that encountered an error that can be recovered from, wdyt @jinqi7 ?

Reply:
@digim, it sounds good for me. It can seperate the snapshots which is marked as invalid and stopped with that ones  which is marked as invalid and requeued.  Will you suggest a name of  the status?

Reply:
Wdyt about a more generic `ErrorOccured`  to have some commonality with the other resources like e.g. in [gitOps here](https://github.com/redhat-appstudio/managed-gitops/blob/ac1312cc4eea936ea9ebe4845b9a48ba5c9d0474/appstudio-controller/controllers/appstudio.redhat.com/environment_controller.go#L265-L268)?

Reply:
Ack, thanks for following up on this